flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/RecordWriter.java

```diff
@@ -54,23 +54,17 @@
  *
  * @param <T> the type of the record that can be emitted with this record writer
  */
-public class RecordWriter<T extends IOReadableWritable> {
+public abstract class RecordWriter<T extends IOReadableWritable> {
 
 	private static final Logger LOG = LoggerFactory.getLogger(RecordWriter.class);
 
-	private final ResultPartitionWriter targetPartition;
+	protected final ResultPartitionWriter targetPartition;
 
-	private final ChannelSelector<T> channelSelector;
+	protected final int numberOfChannels;
```

Comment:
any reason to switch the order of fields here?

Reply:
It was not my intention for switching the order here. I only changed the `numberOfChannels` as `protected` and added one more field `RecordSerializer<T> serializer` below.

I guess if I added the `serializer` field in other regions it would avoid this issue. I would have a try.